### PR TITLE
Add Kolochu and Rumia

### PR DESCRIPTION
**Pull request by ausmacon**
let me know if anything looks fishy here, namely the conditions.

### CODE REVIEW

**Comment by ausmacon** [comment]
just added rumia too but idk how to make it a separate pull req, seems it's adding it to the one i had earlier

**Comment by Karthik99999** [comment]
you need to make a new branch

File: data/mods/ssb/abilities.ts [hunk 1]
```diff
@@ -197,7 +197,23 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
...
-
+	// Kolohe
+	soulsurfer: {
+		name: "Soul Surfer",
+		shortDesc: "Rain on entry; Speed: x2 in Electric Terrain.",
+		onStart(source) {
+			for (const action of this.queue) {
+				if (action.choice === 'runPrimal' && action.pokemon === source && source.species.id === 'kyogre') return;
+				if (action.choice !== 'runSwitch' && action.choice !== 'runPrimal') break;
+			}
```
Comment (Karthik99999) [inline data/mods/ssb/abilities.ts:208]:
```suggestion
```
unnecessary since this is only for primal kyogre

File: data/mods/ssb/conditions.ts [hunk 2]
```diff
@@ -197,6 +197,27 @@ export const Conditions: {[k: string]: ModdedConditionData & {innateName?: strin
...
+			if (source && source.name === 'Rumia') {
+				this.add(`c:|${getName('kolochu ✮彡')}|You come around here often?`);
+			} else if (source && source.name === 'spoo') {
+				this.add(`c:|${getName('kolochu ✮彡')}|Big bald head spotted...`);
+			} else if (source && source.name === 'ausma') {
+				this.add(`c:|${getName('kolochu ✮彡')}|The weekly Smogon furry convention starts NOW`);
+			} else if (source && source.name === 'Peary') {
+				this.add(`c:|${getName('kolochu ✮彡')}|Any arters or culturers?`);
+			} else this.add(`c:|${getName('kolochu ✮彡')}|Hey, howzit!`);
+		},
```
Comment (Karthik99999) [inline data/mods/ssb/conditions.ts:212]:
```suggestion
		onStart(pokemon) {
			const foe = pokemon.foes()[0];
			if (foe && foe.name === 'Rumia') {
				this.add(`c:|${getName('kolochu ✮彡')}|You come around here often?`);
			} else if (foe && foe.name === 'spoo') {
				this.add(`c:|${getName('kolochu ✮彡')}|Big bald head spotted...`);
			} else if (foe && foe.name === 'ausma') {
				this.add(`c:|${getName('kolochu ✮彡')}|The weekly Smogon furry convention starts NOW`);
			} else if (foe && foe.name === 'Peary') {
				this.add(`c:|${getName('kolochu ✮彡')}|Any arters or culturers?`);
			} else {
				this.add(`c:|${getName('kolochu ✮彡')}|Hey, howzit!`);
			}
		},
```

File: data/mods/ssb/conditions.ts [hunk 3]
```diff
@@ -245,6 +266,24 @@ export const Conditions: {[k: string]: ModdedConditionData & {innateName?: strin
...
+		onSwitchOut(source) {
+			if (source && source.name === 'Kolohe') {
+				this.add(`c:|${getName('Rumia')}|i cant bring myself to do this...`);
+			} else this.add(`c:|${getName('Rumia')}|brb ^_^`);
+		},
+		onFaint(source) {
+			if (source && source.name === 'Kolohe') {
+				this.add(`c:|${getName('Rumia')}|this is the best way to go out...`);
+			} else this.add(`c:|${getName('Rumia')}|is that sooooo...`);
+		},
```
Comment (Karthik99999) [inline data/mods/ssb/conditions.ts:288]:
```suggestion
		onStart(pokemon) {
			const foe = pokemon.foes()[0];
			if (foe && foe.name === 'Kolohe') {
				this.add(`c:|${getName('Rumia')}|OMG who could that be (⁠●⁠♡⁠∀⁠♡⁠)`);
			} else {
				this.add(`c:|${getName('Rumia')}|is the mon in front of me the edible kind?`);
			}
		},
		onSwitchOut(pokemon) {
			const foe = pokemon.foes()[0];
			if (foe && foe.name === 'Kolohe') {
				this.add(`c:|${getName('Rumia')}|i cant bring myself to do this...`);
			} else {
				this.add(`c:|${getName('Rumia')}|brb ^_^`);
			}
		},
		onFaint(pokemon) {
			const foe = pokemon.foes()[0];
			if (foe && foe.name === 'Kolohe') {
				this.add(`c:|${getName('Rumia')}|this is the best way to go out...`);
			} else {
				this.add(`c:|${getName('Rumia')}|is that sooooo...`);
			}
		},
```

File: data/mods/ssb/moves.ts [hunk 4]
```diff
@@ -323,7 +323,34 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+		name: "Hang Ten",
+		shortDesc: "User sets Electric Terrain on hit.",
+		pp: 10,
+		priority: 0,
+		flags: {protect: 1, mirror: 1},
+		volatileStatus: 'protect',
+		onPrepareHit(pokemon) {
+			this.add('-anim', source, 'Stoked Sparksurfer', target);
+			this.add('-anim', source, 'Surf', target);
+		},
```
Comment (Karthik99999) [inline data/mods/ssb/moves.ts:341]:
```suggestion
		onTryMove() {
			this.attrLastMove('[still]');
		},
		onPrepareHit(pokemon) {
			this.add('-anim', source, 'Stoked Sparksurfer', target);
			this.add('-anim', source, 'Surf', target);
		},
```

File: data/mods/ssb/moves.ts [hunk 5]
```diff
@@ -323,7 +323,34 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+	hangten: {
+		accuracy: 100,
+		basePower: 75,
+		category: "Special",
+		name: "Hang Ten",
+		shortDesc: "User sets Electric Terrain on hit.",
+		pp: 10,
+		priority: 0,
+		flags: {protect: 1, mirror: 1},
+		volatileStatus: 'protect',
```
Comment (Karthik99999) [inline data/mods/ssb/moves.ts:337]:
```suggestion
```

**Comment by Karthik99999** [comment]
everything else seems ok. just remove rumia from this branch and put the changes on a new branch so that can be a separate pr

File: data/mods/ssb/conditions.ts [hunk 6]
```diff
@@ -197,6 +197,30 @@ export const Conditions: {[k: string]: ModdedConditionData & {innateName?: strin
 			this.add(`c:|${getName('Kennedy')}|LOOOOOOL ffs`);
 		},
 	},
+	kolohe: {
```
Comment (HisuianZoroark) [inline data/mods/ssb/conditions.ts:200]: Rename all traces of Kolohe to be their PS name Kolochu

File: data/mods/ssb/conditions.ts [hunk 7]
```diff
@@ -245,6 +269,33 @@ export const Conditions: {[k: string]: ModdedConditionData & {innateName?: strin
 			this.add(`c:|${getName('phoopes')}|Jynx! Knock on wood`);
 		},
 	},
+	rumia: {
+		noCopy: true,
+		onStart(pokemon) {
+			const foe = pokemon.foes()[0];
+			if (foe && foe.name === 'Kolohe') {
```
Comment (HisuianZoroark) [inline data/mods/ssb/conditions.ts:276]: you can do foe?.name to condense this condition

File: data/mods/ssb/abilities.ts [hunk 8]
```diff
@@ -367,7 +379,22 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
 		name: "I Did It Again",
 		// implemented in rulesets.ts
 	},
-
+	
+	// Rumia
+	youkaiofthedusk: {
```
Comment (HisuianZoroark) [inline data/mods/ssb/abilities.ts:384]: this prankster part will need a dark-type immunity check, will add this when I get home

File: data/mods/ssb/pokedex.ts [hunk 9]
```diff
@@ -36,7 +36,14 @@ export const Pokedex: {[k: string]: ModdedSpeciesData} = {
 		abilities: {0: "You'll Never Walk Alone"},
 		weightkg: 103,
 	},
-
+	
+	// Kolohe
+	pikachu: {
+		inherit: true,
+		baseStats: {hp: 45, atk: 80, def: 50, spa: 75, spd: 60, spe: 120},
+		abilities: {0: "Soul Surfer"},
```
Comment (HisuianZoroark) [inline data/mods/ssb/pokedex.ts:44]: this is not needed, changing abilities in pokédex is only for forme changes

File: data/mods/ssb/conditions.ts (hunk 6 above)
Comment (HisuianZoroark) [reply data/mods/ssb/conditions.ts:200]: PS names MUST be used for Staff Bros. No exceptions unless you have that PS username

**Comment by HisuianZoroark** [comment]
This is fine for now with 2 sets since they’re both p simple but yeah please only 1 set per pull request

**Pull request merged**